### DISCUSSION TITLE
Performance problem with dependency evaluation

### DISCUSSION THREAD

**Discussion by kristjanvalur** [fastapi/fastapi#8402] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# does not apply
```

### Description

Debugging a FastAPI application and the overhead introduced by adding `Depends` type dependencies to endpoints led me to discover problems in the way dependencies are solved.  Please see PR #5554 for information.
That PR removes a _performance bug_ in `solve_dependencies` as well as speeding up non-path parameters for endpoints.
I'm adding this defect for visibilty

### Operating System

Windows

### FastAPI Version

"main" branch

### Python Version

3.10

**Comment by ranjanprasad96** [comment]
@tiangolo Any updates on this issue?
